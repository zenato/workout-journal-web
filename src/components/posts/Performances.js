import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import { Button } from 'components/form';
import Performance from 'components/posts/Performance';
import styles from './PostForm.scss';

const cx = classNames.bind(styles);

const Performances = ({ fields, meta: { error, submitFailed }, events, formValues }) => {
  return (
    <div className={cx('item')}>
      <label>Perf.</label>
      <div className={cx('field')}>
        <Button onClick={() => fields.push({})} value="Add" />
        <table>
          <colgroup>
            <col width="*" />
            <col width="8%" />
            <col width="5%" />
            <col width="8%" />
            <col width="8%" />
            <col width="8%" />
            <col width="8%" />
            <col width="8%" />
            <col width="8%" />
            <col width="5%" />
          </colgroup>
          <thead>
          <tr>
            <th>Name</th>
            <th colSpan={2}>Val.</th>
            <th>1</th>
            <th>2</th>
            <th>3</th>
            <th>4</th>
            <th>5</th>
            <th>Tot. / Vol.</th>
            <th></th>
          </tr>
          </thead>
          <tbody>
          {fields.map((name, index) => (
            <Performance
              key={index}
              name={name}
              events={events}
              values={formValues.performances[index] || {}}
              onDelete={() => fields.remove(index)}
            />
          ))}
          </tbody>
        </table>
      </div>
    </div>
  )
};

Performances.propTypes = {
  fields: PropTypes.shape({
    map: PropTypes.func.isRequired,
    remove: PropTypes.func.isRequired,
  }).isRequired,
  events: PropTypes.array.isRequired,
  formValues: PropTypes.shape({
    performances: PropTypes.array.isRequired,
  }).isRequired,
};

export default Performances;
