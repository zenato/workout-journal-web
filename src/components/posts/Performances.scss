@import 'common';

.performance {
  border: solid 1px #d5d5d5;
  padding: 10px;
  margin-top: 10px;

  .index {
    color: #ca2a00;
    font-weight: bold;
  }
}
