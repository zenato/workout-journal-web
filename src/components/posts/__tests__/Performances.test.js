import React from 'react';
import { shallow } from 'enzyme';
import Performances from '../Performances';

const events = [
  { id: 'key-1', name: 'event-1' },
];

const fields = {
  map: func => [func('performances[0]', 0)],
  remove: () => {}
};

const formValues = {
  performances: [],
};

it('renders without errors', () => {
  const wrapper = shallow(
    <Performances
      key={0}
      fields={fields}
      events={events}
      meta={{}}
      formValues={formValues}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
