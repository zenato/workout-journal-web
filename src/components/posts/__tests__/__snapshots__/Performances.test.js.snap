// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without errors 1`] = `
<div
  className="item"
>
  <label>
    Perf.
  </label>
  <div
    className="field"
  >
    <Button
      onClick={[Function]}
      value="Add"
    />
    <table>
      <colgroup>
        <col
          width="*"
        />
        <col
          width="8%"
        />
        <col
          width="5%"
        />
        <col
          width="8%"
        />
        <col
          width="8%"
        />
        <col
          width="8%"
        />
        <col
          width="8%"
        />
        <col
          width="8%"
        />
        <col
          width="8%"
        />
        <col
          width="5%"
        />
      </colgroup>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th
            colSpan={2}
          >
            Val.
          </th>
          <th>
            1
          </th>
          <th>
            2
          </th>
          <th>
            3
          </th>
          <th>
            4
          </th>
          <th>
            5
          </th>
          <th>
            Tot. / Vol.
          </th>
          <th />
        </tr>
      </thead>
      <tbody>
        <Performance
          events={
            Array [
              Object {
                "id": "key-1",
                "name": "event-1",
              },
            ]
          }
          key="0"
          name="performances[0]"
          onDelete={[Function]}
          values={Object {}}
        />
      </tbody>
    </table>
  </div>
</div>
`;
